src/main/java/de/unijena/cheminf/mortar/controller/HistogramViewController.java

```diff
@@ -59,458 +67,944 @@
-                this.histogramChart.setVerticalGridLinesVisible(false);
-                this.histogramChart.setHorizontalGridLinesVisible(false);
-            }
+        this.histogramView.getDisplayGridLinesCheckBox().selectedProperty()
+                .addListener((ObservableValue<? extends Boolean> ov, Boolean old_val, Boolean new_val) -> {
+            this.histogramChart.setVerticalGridLinesVisible(new_val);
+            this.histogramChart.setHorizontalGridLinesVisible(new_val);
+            //update setting for persistence
+            this.displayGridLinesSetting.set(new_val);
+        });
+        this.histogramView.getDisplaySmilesOnYAxisCheckBox().selectedProperty()
+                .addListener((ObservableValue<? extends Boolean> ov, Boolean old_val, Boolean new_val) -> {
+            this.categoryAxis.setTickMarkVisible(new_val);
+            this.categoryAxis.setTickLabelsVisible(new_val);
+            //update setting for persistence
+            this.displaySMILESSetting.set(new_val);
+        });
+        this.histogramScene.widthProperty().addListener((observable, oldValue, newValue) -> {
+            double tmpWidthChange = ((this.histogramScene.getWidth() - GuiDefinitions.GUI_MAIN_VIEW_WIDTH_VALUE) / GuiDefinitions.GUI_MAIN_VIEW_WIDTH_VALUE) * 100;
+            double tmpImageWidthChange = (HistogramViewController.STRUCTURE_DEPICTION_IMAGE_INITIAL_WIDTH / 100) * tmpWidthChange;
+            this.imageWidth = HistogramViewController.STRUCTURE_DEPICTION_IMAGE_INITIAL_WIDTH + tmpImageWidthChange;
+            this.imageHeight = this.imageWidth - 100;
+            this.imageZoomFactor = (HistogramViewController.STRUCTURE_DEPICTION_IMAGE_INITIAL_ZOOM_FACTOR / HistogramViewController.STRUCTURE_DEPICTION_IMAGE_INITIAL_WIDTH) * this.imageWidth;
         });
-        this.histogramView.getSmilesTickLabel().selectedProperty().addListener((ObservableValue<? extends Boolean> ov, Boolean old_val, Boolean new_val) -> {
-            if (this.histogramView.getSmilesTickLabel().isSelected()) {
-                this.categoryAxis.setTickMarkVisible(false);
-                this.categoryAxis.setTickLabelsVisible(false);
+        this.histogramScene.heightProperty().addListener((observable, oldValue, newValue) -> {
+            double tmpHeightChange = ((this.histogramScene.getHeight() - GuiDefinitions.GUI_MAIN_VIEW_HEIGHT_VALUE) / GuiDefinitions.GUI_MAIN_VIEW_HEIGHT_VALUE) * 100;
+            double tmpImageHeightChange = (HistogramViewController.STRUCTURE_DEPICTION_IMAGE_INITIAL_HEIGHT / 100) * tmpHeightChange;
+            if (this.histogramScene.getWidth() == GuiDefinitions.GUI_MAIN_VIEW_WIDTH_VALUE) {
+                this.imageHeight = HistogramViewController.STRUCTURE_DEPICTION_IMAGE_INITIAL_HEIGHT + tmpImageHeightChange;
             } else {
-                this.categoryAxis.setTickMarkVisible(true);
-                this.categoryAxis.setTickLabelsVisible(true);
+                double tmpHeight = this.imageWidth - 100;
```

Comment:
So with the knowledge I've only had for a short time now, I would say that here (and also in many other places in the class) the numbers have to be given concretely as doubles, right? 
This is of course all my fault.

Reply:
(@FelixBaensch tagging you because you raised this issue)
Yes, I guess this should be "double tmpHeight = this.imageWidth - 100.0", for example.